### ISSUE TITLE
updateHostsFile.py fails to download from KADhosts under Python 2.7

### ISSUE THREAD

**Issue by StevenBlack** [StevenBlack/hosts#372]
Seeing this under Python 2.7.

     Error in updating source:  https://raw.githubusercontent.com/azet12/KADhosts/master/KADhosts.txt

I suspect that [`write_data`](https://github.com/StevenBlack/hosts/blob/2bb7c27269edfa503fb2ce251e0c721dea288eb4/updateHostsFile.py#L889-L904) is failing because of newly added cyrillic characters in commit 2bb7c27269edfa503fb2ce251e0c721dea288eb4.

**Comment by ARROWBOW** [comment]
Unable to use it.Not working.

**Comment by meinhimmel** [comment]
Having the same issue with Python 3.6.2 on macOS 10.12.6.

```
python3 updateHostsFile.py --auto
```

```
Updating source data/adaway.org from https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt
Updating source data/add.2o7Net from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.2o7Net/hosts
Updating source data/add.Dead from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Dead/hosts
Updating source data/add.Risk from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Risk/hosts
Updating source data/add.Spam from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Spam/hosts
Updating source data/Badd-Boyz-Hosts from https://raw.githubusercontent.com/mitchellkrogza/Badd-Boyz-Hosts/master/hosts
Updating source data/KADhosts from https://raw.githubusercontent.com/azet12/KADhosts/master/KADhosts.txt
Updating source data/malwaredomainlist.com from https://www.malwaredomainlist.com/hostslist/hosts.txt
Updating source data/mvps.org from http://winhelp2002.mvps.org/hosts.txt
Traceback (most recent call last):
  File "updateHostsFile.py", line 1253, in <module>
    main()
  File "updateHostsFile.py", line 156, in main
    settings["hostfilename"])
  File "updateHostsFile.py", line 513, in update_all_sources
    update_data = json.load(update_file)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py", line 296, in load
    return loads(fp.read(),
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 27: ordinal not in range(128)
```

**Comment by funilrys** [comment]
Gonna try to fix it once I have time but if someone else have time to write a PR:

Here's in my opinion the best way to achieve the string decoding/encoding: https://stackoverflow.com/a/15918519/5308363

**Comment by FadeMind** [comment]
@StevenBlack  FIXED in https://github.com/azet12/KADhosts/commit/0649a5b745241093c9e9b0df8555318b8e93063b

**Comment by StevenBlack** [comment]
Thanks Tomasz @FadeMind.  

We still need to fix the encoding issue on Python 2.7...

**Comment by funilrys** [comment]
You can also close this @StevenBlack :+1:
